packages/wasm/crate/tests/test_build.rs

```diff
@@ -1,577 +1,398 @@
-        };
-
-        let constants: IndexedDbConstants = IndexedDbConstants {
-            name: "penumbra-db-wasm-test".to_string(),
-            version: 1,
-            tables,
-        };
-
-        // Sample chain and fmd parameters.
-        let chain_id = "penumbra-testnet-iapetus".to_string();
-        let sct_params = SctParameters {
-            epoch_duration: 5u64,
-        };
-        let dex_params = DexParameters {
-            fixed_candidates: Vec::new(),
-            is_enabled: true,
-            max_hops: 5u32,
-            max_positions_per_pair: 0,
-            max_execution_budget: 0u32,
-        };
-
-        let app_params = AppParameters {
-            dex_params: Some(dex_params.to_proto()),
-            chain_id,
-            sct_params: Some(sct_params.to_proto()),
-            community_pool_params: None,
-            governance_params: None,
-            ibc_params: None,
-            stake_params: None,
-            fee_params: None,
-            distributions_params: None,
-            funding_params: None,
-            shielded_pool_params: None,
-            auction_params: None,
-        };
-
-        let fmd_params = FmdParameters {
-            precision_bits: 0u32,
-            as_of_block_height: 1u64,
-        };
-        let gas_prices = GasPrices {
-            asset_id: Some((*STAKING_TOKEN_ASSET_ID).into()),
-            block_space_price: 0,
-            compact_block_space_price: 0,
-            verification_price: 0,
-            execution_price: 0,
-        };
-
-        // Serialize the parameters into `JsValue`.
-        let js_app_params_value: JsValue = serde_wasm_bindgen::to_value(&app_params).unwrap();
-        let js_fmd_params_value: JsValue = serde_wasm_bindgen::to_value(&fmd_params).unwrap();
-        let js_constants_params_value: JsValue = serde_wasm_bindgen::to_value(&constants).unwrap();
-        let js_gas_prices_value: JsValue = serde_wasm_bindgen::to_value(&gas_prices).unwrap();
-
-        // Create spendable UTXO note in JSON format.
-        let spendable_note_json = r#"
-        {
-            "note_commitment": {
-                "inner": "MY7PmcrH4fhjFOoMIKEdF+x9EUhZ9CS/CIfVco7Y5wU="
-            },
-            "note": {
-                "value": {
-                    "amount": {
-                        "lo": "1000000",
-                        "hi": "0"
-                    },
-                    "asset_id": {
-                        "inner": "nwPDkQq3OvLnBwGTD+nmv1Ifb2GEmFCgNHrU++9BsRE=",
-                        "alt_bech32m": "",
-                        "alt_base_denom": ""
-                    }
+    // Create spendable UTXO note in JSON format.
+    let spendable_note_json = r#"
+    {
+        "note_commitment": {
+            "inner": "MY7PmcrH4fhjFOoMIKEdF+x9EUhZ9CS/CIfVco7Y5wU="
+        },
+        "note": {
+            "value": {
+                "amount": {
+                    "lo": "1000000",
+                    "hi": "0"
```

Comment:
Maintaining these strings manually feels like a chore. Can we just test the inner function of the build and not feel like we need to test the deserialization logic?

Reply:
replaced the hardcoded json with their corresponding typed structures.